lib/phoenix-endpoint-campaigns.js

```diff
@@ -0,0 +1,45 @@
+'use strict';
+
+/**
+ * Imports.
+ */
+const PhoenixEndpoint = require('./phoenix-endpoint');
+
+
+/**
+ * PhoenixEndpointCampaigns.
+ */
+
+class PhoenixEndpointCampaigns extends PhoenixEndpoint {
+
+  constructor(client) {
+    super(client);
+    this.endpoint = 'campaigns';
+  }
+
+  /**
+   * Post Signup for given Campaign id and data.
+   */
+  signup(id, data) {
+    return this
+      .callAction('signup', id, data)
+      .then((response) => {
+        const body = this.parseStringResponse(response);
+        return Number(body);
```

Comment:
The reason why your test doesn't fail is that `body` for the test is actually returned `false`.
Thus `Number('false')` [evaluates](**url**) to `NaN`.

However, [`NaN`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/NaN) (stands for "not a number") **is**, in fact, a `Number`. Javascript :trollface:.

So it's totally valid to return a `Number` from Promise. We need to check the result for `NaN` and treat it as an error, throw an exception (see `parseResponse()`).